libs/vkd3d/state.c

```diff
@@ -343,42 +370,68 @@ struct d3d12_root_signature_info
 };
 
 static HRESULT d3d12_root_signature_info_count_descriptors(struct d3d12_root_signature_info *info,
-        const D3D12_DESCRIPTOR_RANGE *range)
+        struct d3d12_device *device, const D3D12_DESCRIPTOR_RANGE *range)
 {
     size_t descriptor_count;
 
-    if (range->NumDescriptors == 0xffffffff)
+    if (d3d12_root_signature_descriptor_range_is_bindless(device, range->RangeType))
     {
-        FIXME("Unhandled unbound descriptor range.\n");
-        return E_NOTIMPL;
+        switch (range->RangeType)
+        {
+            case D3D12_DESCRIPTOR_RANGE_TYPE_SRV:
+                /* separate image + buffer descriptors */
+                info->binding_count += 2;
```

Comment:
Probably best to set descriptor_count here and increment outside the branch. The two branches here to the same thing in different ways.

Reply:
The difference is that bindings and packed descriptors are decoupled here. Can of course introduce a separate `binding_count`, but this will lead to some duplication in the second branch.

Reply:
Hm, right, nvm.